microcosm_secretsmanager/loaders/base.py

```diff
@@ -0,0 +1,63 @@
+from json import dumps, loads
+from json.decoder import JSONDecodeError
+
+from boto3 import Session
+from botocore.exceptions import ClientError
+
+from microcosm_logging.decorators import logger
+
+@logger
+class SecretsManagerLoader:
+    def __init__(self,
+                 environment=None,
+                 profile_name=None,
+                 region=None):
+
+        self.environment = environment
+        self.profile_name = profile_name
+        self.region = region
+
+    def __call__(self, metadata, version=None):
+        service = metadata if isinstance(metadata, str) else metadata.name
+        return self.get_secret_value(service, version)
+
+
+    def get_secret_value(self, service, version=None):
+        keyname = self.keyname(service)
+        client = self._client(service)
+
+        get_args = dict(
+            SecretId=keyname,
+        )
+
+        if version:
+            get_args.update(
+                VersionStage=version,
+            )
+
+        try:
+            response = client.get_secret_value(
+                **get_args,
+            )
+
+        except ClientError as e:
+            self.logger.error(f"Error from Amazon retrieving the secrets for: {keyname}, most likely resource does not exist")
```

Comment:
If you are returning `{}`, this not an error because you're designing this to be non-fatal.

If you re-raised, I'd accept this as a valid *error* log because it implies a systematic failure. That said, we probably won't be able to write this error log to anywhere interesting in the likely event that remote logging requires a secret.

So maybe re-raise and just `print`?